monitoring/rid_qualifier/flight_track_factory.py

```diff
@@ -0,0 +1,178 @@
+            # Get the bounds of the buffered box, this is the one that will will be fed to the remote ID display provider to query
+            self.query_bboxes.append({'name':box_diagonals[box_id]['name'], 'shape': buffered_box})
+            
+
+    def generate_flight_grid(self):
+        ''' Generate a series of boxes within the bounding box to have areas for different flights '''
+        # Compute the box where the flights will be created. For a the sample bounds given, over Bern, Switzerland, a division by 2 produces a cell_size of 0.0025212764739985793, a division of 3 is 0.0016808509826657196 and division by 4 0.0012606382369992897. As the cell size goes smaller more number of flights can be accomodated within the grid. For the study area bounds we build a 3x2 box for six flights by creating 3 column 2 row grid. 
+
+        cell_size_x = (self.maxx - self.minx)/(3) # create three columns
+        cell_size_y = (self.maxy - self.miny)/(2) # create two rows
+        grid_cells = [] 
+        for u0 in range(0, 3):  # 3 columns           
+            x0 =  self.minx + (u0 * cell_size_x)            
+            for v0 in range(0,2): # 2 rows
+                y0 = self.miny + (v0 *cell_size_y)
+                x1 = x0 + cell_size_x
+                y1 = y0 + cell_size_y
+                grid_cells.append(shapely.geometry.box(x0, y0, x1, y1))                
+        
+        self.flight_grid = grid_cells
+        
+
+    def generate_flight_paths_points(self):
+        ''' For each of the boxes allocated to the operator, get the centroid and buffer to get a flight path. A 75 m radius is provided to have flight paths within each of the boxes '''
+        # Iterate over the flight_grid
+        for grid_cell in self.flight_grid:
+            center = grid_cell.centroid
+            ## Transfrom to buffer 100 m diameter circle on which the drone will fly            
+            transformer = Transformer.from_crs("epsg:4326", "epsg:3857")
+            transformed_x,transformed_y = transformer.transform(center.x, center.y)
+            pt = Point(transformed_x,transformed_y)
+            buffer = pt.buffer(70) # build a buffer so that the radius is 70m for the track
+            buffer_points =  zip(buffer.exterior.coords.xy[0], buffer.exterior.coords.xy[1])
+            proj_buffer_points = []
+            # reproject back to ESPG 4326
+            transformer2 = Transformer.from_crs("epsg:3857", "epsg:4326")
+            for point in buffer_points:
+                x = point[0]
+                y = point[1]
+                x, y = transformer2.transform(x, y)
+                proj_buffer_points.append((x, y))
+            buffered_path = Polygon(proj_buffer_points)
+            altitude = 70.0
+            flight_points_with_altitude = []
+            x, y = buffered_path.exterior.coords.xy
+            for coord in range(0,len(x)):
+                flight_points_with_altitude.append((x[coord], y[coord], altitude))
```

Comment:
I think it will be very important to attach times to each of these points.  In addition to being necessary for auditing purposes (understanding what aircraft trajectories, including speeds, are being injected into the systems), it will also be necessary to synchronize test queries with expected aircraft flight states.  For instance, the last query of the Off-nominal behavior test should happen with just one flight left in the air -- we'll need to know that we should expect those conditions to be present between, e.g., T+2:00 and T+2:45 from the reference time.

I think even better would be if we simply populate the full information for [RIDAircraftState](https://github.com/uastech/standards/blob/36e7ea23a010ff91053f82ac4f6a9bfc698503f9/remoteid/canonical.yaml#L1604) for each point as that's what aircraft_state_replayer will need to know.

Reply:
OK no problems, my thinking is that since this is a circular path, we have them flying for a long time looping overt the tracks. I will add a time_step in seconds. I will update the structure to the RID Aircraft State here itself. What I was thinking is that the `flight_data_generator` will just produce the tracks and the `state_replayer` will produce the aircraft state (from the tracks) depending on how long we want them "fly" and then create the Aircraftstate from the tracks . But you are suggesting that we create the aircraft state here itself. I will add a method here and will refactor the write geojson as well.

Reply:
I will resolve this for the moment, we can re-visit it in the next iteration of the review.

Reply:
While it's not important to do every part of the implementation or get all the implementation details perfect initially, I do think it's important to be very clear about interfaces throughout the development process as they tend to drive direction of development.  As described [here](https://github.com/interuss/dss/pull/483#discussion_r585916979), I think it's important to separate "simulation" and "replay", based on feedback from people who have designed and qualified other aviation test systems.  I think creating time stamps from scratch is too much to do in the "replay" block because that would essentially be simulating the aircrafts' speeds during the flight.  From an audit perspective, the audit would be unable to verify that unreasonable speeds were not being fed into the system if we apply timestamps after audit, so that is one dimension in which the audit couldn't be fully completed.

From a longer view, I think we're likely to want to create more complicated flight patterns in the future.  We don't need them now, but we probably will need them to reliably test, e.g., behavior when an aircraft repeatedly enters and exits a view within the past 60 seconds.  To prepare for that future change, we should set up our interfaces now to easily support that future use case.  As mentioned at the root of this thread, I think the most robust approach would be to output verbatim RIDAircraftStates, with the only modification being a time delta rather than a timestamp (since timestamps must necessarily be dynamically modified by the aircraft state replayer).  I think an acceptable compromise for now may be to output only (x,y,z,t), but I think (x,y,z) is insufficient as it will drive computation of t into the aircraft state replayer which does not match our desired split of simulation and replay.

Reply:
Thanks, yes agreed I have made the updates but just slightly differently, I add the time stamp at the very end and keep the aircraft position data with out time (per the API). I will send on a update for your review shortly.